src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -614,14 +614,13 @@ void GameObject::Update(uint32 diff)
 
             //! If this is summoned by a spell with ie. SPELL_EFFECT_SUMMON_OBJECT_WILD, with or without owner, we check respawn criteria based on spell
             //! The GetOwnerGUID() check is mostly for compatibility with hacky scripts - 99% of the time summoning should be done trough spells.
-            if (GetSpellId() || GetOwnerGUID())
+            if ((GetSpellId() || GetOwnerGUID()) && m_respawnTime > 0)
             {
-                SetRespawnTime(0);
-                Delete();
+                UpdateObjectVisibility();
+                SetLootState(GO_READY);
                 return;
             }
-
-            SetLootState(GO_READY);
+            else Delete();
```

Comment:
the old code returned after calling Delete(), is there any difference now not returning anymore ?